shell/src/main/java/alluxio/cli/docgen/ConfigurationDocValidator.java

```diff
@@ -0,0 +1,176 @@
+                                       String fileDir, List<String> fileNames)
+          throws IOException {
+
+    String filePath = PathUtils.concatPath(Configuration.getString(PropertyKey.HOME), fileDir);
+    Integer numDeltas = 0;
+    for (String fileName : fileNames) {
+      String key = fileName.substring(0, fileName.indexOf("configuration") - 1);
+
+      // based on key substring within each filename, create a regex to
+      // filter for specific property keys.
+      List<String> rPatterns = new ArrayList<String>();
+      switch (key) {
+        case "user":
+          rPatterns.add("^alluxio\\.(user|user\\..*)");
+          break;
+        case "master":
+          rPatterns.add("^alluxio\\.(master|master\\..*)");
+          break;
+        case "worker":
+          rPatterns.add("^alluxio\\.(worker|worker\\..*)");
+          break;
+        case "security":
+          rPatterns.add("^alluxio\\.(security|security\\..*)");
+          break;
+        case "cluster-management":
+          rPatterns.add("^alluxio\\.(integration|integration\\..*)");
+          break;
+        case "common":
+          rPatterns.add("^alluxio\\.(?!(user|master|worker|security|integration)\\b)\\w*[\\.]?.*");
+          // additional regex pattern to retain non-system configuration records.
+          rPatterns.add("^(fs|s3a)\\b\\..*");
+          break;
+        default:
+          // ignore and skip any non-relevant files.
+          LOG.warn("Ignoring unrecognized filename substring key: " + key);
+          continue;
+      }
+
+      List<Pattern> compiledRegex = new ArrayList<Pattern>();
+      for (String pattern : rPatterns) {
+        compiledRegex.add(Pattern.compile(pattern));
+      }
+
+      List<String> revised = new ArrayList<String>();
+      if (fileDir.equals(CSV_FILE_DIR)) {
+        revised.add(CSV_FILE_HEADER);
+      }
+      for (PropertyKey propertyKey : dfkeys) {
+        String pKey = propertyKey.toString();
+        Boolean matched = false;
+        for (Pattern pattern : compiledRegex) {
+          if (pattern.matcher(pKey).matches()) {
+            matched = true;
+          }
+        }
+        if (matched) {
+          String pKeyDesc;
+          if (fileDir.equals(CSV_FILE_DIR)) {
+            pKeyDesc = getPropertyDescriptionCSV(pKey, propertyKey);
+          } else {
+            pKeyDesc = StringEscapeUtils.escapeHtml4(getPropertyDescriptionYML(pKey, propertyKey));
+          }
+          // need to split multi-line strings with embedded newline character into separate
+          // list records. This is needed to line up records from original files loaded from
+          // the file system.
+          revised.addAll(Arrays.asList(pKeyDesc.split("\n")));
+        }
+      }
+
+      String fileFullPathName = PathUtils.concatPath(filePath, fileName);
+      List<String> original = Files.readAllLines(Paths.get(fileFullPathName));
+
+      numDeltas += generateDiff(original, revised, fileFullPathName);
```

Comment:
i'm confused what all this additional logic to generate the `revised` contents. why is it so different than the generation? couldn't the generation logic be redirected to write to effectively a string in memory, then compare that value with the lines read in the corresponding file?